android/app/src/main/java/app/organicmaps/routing/NavigationService.java

```diff
@@ -46,6 +46,7 @@ public class NavigationService extends Service implements LocationListener
 {
   private static final String TAG = NavigationService.class.getSimpleName();
   private static final String STOP_NAVIGATION = "STOP_NAVIGATION";
+  private static final int REQ_CODE_STOP_NAVIGATION = 1;
```

Comment:
Should these codes be unique compared to MwmActivity ones? If yes, then it would be safer to keep all such codes in a single file/place, to avoid accidentally reusing the same code for different notifications. WDYT?

Reply:
I found another cleaner way to deal with the issue. As all the pending intents do the same thing, one can use FLAG_UPDATE_CURRENT which will not cancel the previous invocation. As a result all the services hold a reference to the same pending intent doing the same thing i.e. opening MwmActivity.

Reply:
I will go with FLAG_UPDATE_CURRENT

As android suggests to use FLAG_CANCEL_CURRENT only if you want to invalidate all other instances of pending intent by any service but here our logic is not to invalidate previous instances and also using it will remove the problem of similar pending intent creation by  mistake.